aptos-move/script-composer/src/builder.rs

```diff
@@ -390,11 +399,11 @@ impl TransactionComposer {
             }
 
             // Storing return values
-            for arg in call.returns {
+            for arg in call.returns.iter().rev() {
                 script
                     .code
                     .code
-                    .push(Bytecode::StLoc((arg + parameters_count) as u8));
+                    .push(Bytecode::StLoc((*arg + parameters_count) as u8));
```

Comment:
Although not part of this PR, do we check for "returns + parameters < 255", or does that happen later when running the script?

Reply:
It will be checked by the bytecode verifier. But I added a check so that the user can be prompted earlier and not having to see the lengthy bytecode verifier error.

Reply:
@runtian-zhou There are a bunch of `as u*` casts in this file, but it is not clear if all of those are safe casts. E.g., `num_of_calls` unconditionally cast to u16.

For example, in the case above, without the check for `u8::MAX`, the value would get silently truncated and have incorrect code generated. It is likely that something downstream (like the bytecode verifier) would catch it, but might be good to add checks here in the builder (unless some of these are checked earlier somewhere).